### PR TITLE
Search/Filter Tokens in OrdersWidget

### LINKED ISSUE #1032: Filter orders/trades
Filter orders / trades.

At least by status, token

### PR DESCRIPTION
**Pull request by W3stside**
Closes #1032 

**Blurred/Unfocused w/placeholder**:
Screenshot from 2020-06-09 14-03-29
**Focused w/text**
Screenshot from 2020-06-09 14-03-21

Opens with `0.2s` open/close transition

### CODE REVIEW

File: src/components/SearchFilter.tsx [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  focusBgColor = 'initial',
+  blurBgColor = 'initial',
+  focusFontColor = 'initial',
+  blurFontColor = 'initial',
+}) => {
+  const [focus, setFocus] = useSafeState(false)
+  const [filter, setFilter] = useSafeState('')
+
+  const handleBlur = useCallback((): void => {
+    batchUpdate(() => {
```
Comment (Velenir) [inline src/components/SearchFilter.tsx:48]: No need to batch updates here.
Updates are batched by default inside sync code in effect hooks and in event handlers

File: src/components/SearchFilter.tsx [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    <SearchFilterInput
+      type="text"
+      placeholder="🔍"
+      value={filter}
+      onChange={handleOnChange}
+      onFocus={(): void => setFocus(true)}
+      onBlur={handleBlur}
+      $background={focus ? focusBgColor : blurBgColor}
+      $fontColor={focus ? focusFontColor : blurFontColor}
+      $width={focus ? openWidth : closedWidth}
```
Comment (Velenir) [inline src/components/SearchFilter.tsx:66]: Wouldn't it be easier to do `:focus` in css?

Comment (W3stside) [reply src/components/SearchFilter.tsx:66]: yes yes this is just prototype stuff just did it all in one file initially

Comment (W3stside) [reply src/components/SearchFilter.tsx:66]: draft AF

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr1092--dexreact.review.gnosisdev.com

File: src/components/SearchFilter.tsx (hunk 1 above)
Comment (W3stside) [reply src/components/SearchFilter.tsx:48]: true!

**Comment by anxolin** [comment]
I like more the design we used in Balances, and I think we should be coherent there.

image

Maybe we can even consider adding the check "[ ] Hide untouched orders"

**Comment by W3stside** [comment]
closing in favour of #1097

**Pull request closed without merging**